django/db/models/fields/composite.py

```diff
@@ -42,18 +39,14 @@ def get_prep_lookup(self):
                 "MultiColSource"
             )
         if not isinstance(self.rhs, (tuple, list)):
-            raise ValueError(
-                "The right-hand side of the 'in' lookup must be a tuple or a list"
-            )
+            raise ValueError(f"'{self.lhs.field.name}' must be a tuple or a list")
         if not all(isinstance(vals, (tuple, list)) for vals in self.rhs):
             raise ValueError(
-                "The right-hand side of the 'in' lookup must be a collection of "
-                "tuples or lists"
+                f"'{self.lhs.field.name}' must be a collection of tuples or lists"
             )
         if not all(len(self.lhs) == len(vals) for vals in self.rhs):
             raise ValueError(
-                "The left-hand side and right-hand side of the 'in' lookup must "
-                "have the same number of elements"
+                f"'{self.lhs.field.name}' must have {len(self.lhs)} elements"
```

Comment:
I don't think this is quite clear. If the `pk` has two component fields, but the lookup is `pk__in=([1, 2], [3])` we will raise an error `'pk' must have 2 elements` but the tuple *does* have two elements. I'm not sure what to suggest instead though - it's tricky...

Reply:
How about this?

> 'exact' lookup of 'primary_key' field must have 2 elements each